src/armor_layers.cpp

```diff
@@ -439,8 +444,17 @@ void player::sort_armor()
     * because some items can have multiple entries (i.e. cover a few parts of body).
     */
 
-    int req_right_h = 3 + 1 + 2 + num_bp + 1;
-    for( const bodypart_id &cover : get_all_body_parts() ) {
+    // FIXME: get_all_body_parts() doesn't return a sorted list
+    //        and bodypart_id is not compatible with std::sort()
+    //        so let's use a dirty hack
+    cata::flat_set<bodypart_id> armor_cat;
+    for( const auto &it : get_all_body_parts() ) {
+        armor_cat.insert( it );
+    }
+    armor_cat.insert( bodypart_id( body_part::num_bp ) );
```

Comment:
```suggestion
    armor_cat.insert( bodypart_id( "num_bp" ) );
```
isn't num_bp already returned by get_all_body_parts()?